### PR TITLE
Two fixes for MinGW64, and one for non-root builds

### PR DESCRIPTION
**Pull request by wiiaboo**
Great software, use it a lot. Keep it up 👍

### CODE REVIEW

File: src/builtin.c [hunk 1]
```diff
@@ -34,6 +34,9 @@ void *alloca (size_t);
 #endif
 #include <string.h>
 #include <time.h>
+#if defined(WIN32) && !defined(HAVE_SETENV)
```
Comment (nicowilliams) [inline src/builtin.c]: Make that just `#ifdef WIN32`.  Whether we have `setenv()` is of no relevance here, right?

File: src/lexer.l [hunk 2]
```diff
@@ -1,7 +1,7 @@
 %{
 #include <assert.h>
-#include "jv_alloc.h"
-#include "compile.h"
+#include "src/jv_alloc.h"
```
Comment (nicowilliams) [inline src/lexer.l]: Oh?  `-I${srcdir}` is not enough?

Comment (wiiaboo) [reply src/lexer.l]: I ran make with `V=1` and sure enough, there were just `-I. -I..`. The `..` is the `-I${srcdir}`, not `root/src`.

Comment (nicowilliams) [reply src/lexer.l]: I see the same.  Why does jq build for me on Unix when building out of tree?

**Comment by nicowilliams** [comment]
@wiiaboo Thanks for your contribution!  I've rebased it, made the above change, and pushed it.

**Pull request closed without merging**